pipeline/test/end-to-end/test_scoring.jl

```diff
@@ -2,103 +2,38 @@ using DrWatson, Test
-    vline!(plt, [n_observation_steps + missing_padding + 0.5], label = "forecast start")
+    truthdata = generate_truthdata(truth_data_config, pipeline)
 
-    ##Generate true Rt values
-    truth_It = truth_sampling["inference_results"].generated.I_t
-    truth_Rt = expected_Rt(truth_sampling["epiprob"].epi_model.data,
-        truth_sampling["inference_results"].generated.I_t)
-    plt_Rt = plot(truth_Rt, xlabel = "t", ylabel = "R_t", label = "truth Rt")
-    vline!(plt_Rt, [tspan_inf[2] + 0.5], label = "forecast start")
+    ## Set up data generation on a random scenario with reference time in the middle
+    inference_method = make_inference_method(pipeline)
+    inference_config = make_inference_configs(pipeline) |>
+                       cfgs -> filter(cfg -> cfg["T"] == 35, cfgs) |> rand
 
-    ##
-    ## Run inference
-    obs_truthdata = truthdata[tspan_inf[1]:tspan_inf[2]]
+    ## Run inference, forecasting and scoring
 
     inference_results = generate_inference_results(
-        Dict("y_t" => obs_truthdata, "truth_gi_mean" => inference_config["gi_mean"]),
-        inference_config, pipeline; tspan = tspan_inf, inference_method)
-
-    ## Make 21-day forecast
-
-    forecast_quantities = generate_forecasts(inference_results, lookahead)
-    forecast_y_t = mapreduce(hcat, forecast_quantities.generated) do gen
-        gen.generated_y_t
-    end
-    forecast_qs = mapreduce(hcat, [0.025, 0.25, 0.5, 0.75, 0.975]) do q
-        map(eachrow(forecast_y_t)) do row
-            if any(ismissing, row)
-                missing
-            else
-                quantile(row, q)
-            end
-        end
-    end
-    plt = scatter(truthdata, xlabel = "t", ylabel = "y_t", label = "truth data")
-    vline!(plt, [tspan_inf[2] + 0.5], label = "forecast start")
-    plot!(plt, forecast_qs, label = "forecast quantiles",
-        color = :grey, lw = [0.5 1.5 3 1.5 0.5])
-    plot!(plt, ylims = (-0.5, maximum(truthdata) * 1.25))
-    plot!(
-        plt, title = "Forecast of y_t", ylims = (
-            -0.5, maximum(skipmissing(truthdata)) * 1.55))
-    savefig(plt,
-        joinpath(@__DIR__(), "forecast_y_t.png")
-    )
-    display(plt)
-
-    ## Make forecast plot for Z_t
-    infer_Z_t = mapreduce(hcat, inference_results["inference_results"].generated) do gen
-        gen.Z_t
-    end
-    forecast_Z_t = mapreduce(hcat, forecast_quantities.generated) do gen
-        gen.Z_t
+        truthdata, inference_config, pipeline; inference_method,
+        datadir_name = "example_epiaware_observables")
+
+    ## plot scores
+
+    score_results = inference_results["score_results"]
+    targets = filter(ky -> ky != "ts", keys(score_results))
+    ts = score_results["ts"]
+    plt_scores = plot(;
+        xlabel = "Time", ylabel = "CRPS", title = "CRPS Scores (relative to max)")
+    for target in targets
+        scores = score_results[target]
+        plot!(plt_scores, ts, scores ./ maximum(scores), label = target)
```

Comment:
I think we might want to reasses this (scaled to the max) depending on how we want to talk about performance (i.e it doesn't really make sense if we are mostly interested in talking about well performing models.